feat: implement TaskController.saveTask Rest Api (#2677)

igdianov opened the pull request:
Part of Activiti/Activiti#2677

igdianov committed:
feat: implement TaskController.saveTask Rest Api

salaboy commented:
@erdemedeiros please review this one.

erdemedeiros commented on activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-api/src/main/java/org/activiti/cloud/services/rest/api/TaskController.java:
```diff
@@ -37,6 +38,10 @@
 
+    @RequestMapping(value = "/{taskId}/save", method = RequestMethod.POST)
+    void saveTask(@PathVariable String taskId,
+                  @RequestBody(required = false) SaveTaskPayload saveTaskPayload);
```
@igdianov any specific reason to make `SaveTaskPayLoad` optional? I the case of complete task it makes sense because we can complete a task without providing any variable. In case of `save`, it sounds mandatory to me, as calling `save` without payload will have no effect.

igdianov replied:
@erdemedeiros Done. See https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/296/commits/a5cb18c167610470ba083dfb97232033aec3d00c

erdemedeiros commented on activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/test/java/org/activiti/cloud/services/rest/controllers/TaskControllerImplIT.java:
```diff
@@ -209,6 +209,16 @@ public void completeTask() throws Exception {
 
+    @Test
+    public void saveTask() throws Exception {
+        this.mockMvc.perform(post("/v1/tasks/{taskId}/save",
```
We should provide a payload with variables here I think.

igdianov replied:
@erdemedeiros I wast looking into post(..) signature but there is no body argument provided.

igdianov replied:
@erdemedeiros Ok, I found how to set the content body. See: https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/296/commits/66433f0c7483210dcdf0f1ab53de88be1f8a0545

erdemedeiros reviewed (changes requested):
@igdianov see my comment about payload being optional.

salaboy reviewed (changes requested):
@igdianov there are some good comments from @erdemedeiros can we check those so we can merge these PRs?

igdianov committed:
fix: require not empty payload to save a task

igdianov committed:
fix: set content payload in saveTask MockMvc unit test

igdianov committed:
fix: try clean up process instance after task save.

erdemedeiros committed:
Remove commented-reviews-by rule from mergify

This is preventing PRs to get merged even after being approved by all the reviewers

salaboy commented:
I've merged this one because mergify wasn't not working here.
